### PR TITLE
[Enabler][zos_archive][zos_unarchive] Add support for GDGs and validate symbols support

### LINKED ISSUE #1367: [Enabler] [zos_archive] GDG/GDS and special character support
# Enabler description

```python
MVSDataSet(
            name = params.get("name"),
            record_format = params.get("record_format", None),
            volumes = params.get("volumes", None),
            data_set_type = params.get("type", None),
            block_size = params.get("block_size", None),
            record_length = params.get("record_length", None),
            space_primary = params.get("space_primary", None),
            space_secondary = params.get("space_secondary", None),
            space_type = params.get("space_type", None),
            directory_blocks = params.get("directory_blocks", None),
            key_length = params.get("key_length", None),
            key_offset = params.get("key_offset", None),
            sms_storage_class = params.get("sms_storage_class", None),
            sms_data_class = params.get("sms_data_class", None),
            sms_management_class = params.get("sms_management_class", None),
        )
```

GDG/GDS Support
-------------------

`src` option in `zos_archive` accepts a list of string either as a pattern or a list of data set names, so user can provide a list of data set names that contain some GDS relative data set names, in that case, the module needs to resolve the relative name into absolute name before packing to correctly archive the given data set name.

```yaml
- name: Compress data set into a terse, specify pack algorithm and use adrdssu
  zos_archive:
    src: 
      -  "USER.GDG(0)"
      -  "USER.NEWGDG(-2)"
    dest: "USER.ARCHIVE.RESULT.TRS"
    format:
      name: terse
      format_options:
        terse_pack: "SPACK"
        use_adrdssu: True
```

Internally, in the case of a data set name being a GDS relative name we should resolve those names into absolute GDS names when the generation number is 0 or negative.

Following functions will be implemented in #1386 
```python
# Sample of soon to be implemented gdg function
if DataSet.is_gds_relative_name(name):
    src = DataSet.resolve_gds_absolute_name(src)
```

Symbols support
-----------------
As part of the special character support we need to leverage the escape function from the betterArgParser created on #1382 to escape each data set name. This module uses different system utilities like TERSE, TSOCMD, ADRDSSU, **as part of this task we still need to validate if escaping is needed or not in this module**.

Changes done to the BetterArgParser should be adopted in the module to not break and use the correct escaped name in case of the data set name having a symbol in it.

That way, the modules will have to retrieve the new escaped named and the original user provided name inside the dictionary like:

```python
name = DataSet.escape_data_set_name(name)
```

For this task completion we need:
- [ ] Do the development work
- [ ] Update the docs to talk about GDG and GDS handling in the module.
- [ ] Update the docs with at least 1 example on how to use with GDSs.
- [ ] Create a new test case for GDSs.
- [ ] Create a new test case for symbols.

### Ansible module

zos_archive

### PR DESCRIPTION
**Pull request by fernandofloresg**
##### SUMMARY

Add support for GDGs and add tests for special character support.

Fixes #1367 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

zos_archive

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by fernandofloresg** [comment]
zos_archive 
Screenshot 2024-06-11 at 11 22 07 AM

zos_unarchive
Screenshot 2024-06-11 at 11 22 26 AM

File: changelogs/fragments/1367-zos_archive-gdg-support.yml
Comment (richp405) [inline changelogs/fragments/1367-zos_archive-gdg-support.yml:1]: The comment for the PR says this is #1511, for zos_archive, but the fragment doesn't say that.

File: plugins/modules/zos_unarchive.py [hunk 1]
```diff
@@ -35,7 +35,7 @@
       - The remote absolute path or data set of the archive to be uncompressed.
       - I(src) can be a USS file or MVS data set name.
       - USS file paths should be absolute paths.
-      - MVS data sets supported types are C(SEQ), C(PDS), C(PDSE).
```
Comment (richp405) [inline plugins/modules/zos_unarchive.py:38]: Should line 38 be removed?  I thought you added gdg, but didn't remove SEQ/PDS/et cetera.

**Review by richp405 — CHANGES_REQUESTED** [review]
my concern is the changelog fragment: 1 - the link should point to the pull request, 2 - the file needs to mention archive and unarchive.  I see the data set change was probably needed for this to work, so it should be mentioned in the fragment, but after the main changes.

File: changelogs/fragments/1367-zos_archive-gdg-support.yml
Comment (fernandofloresg) [reply changelogs/fragments/1367-zos_archive-gdg-support.yml:1]: sorry forgot to update this.

**Review by AndreMarcel99 — APPROVED** [review]

File: plugins/modules/zos_unarchive.py (hunk 1 above)
Comment (rexemin) [reply plugins/modules/zos_unarchive.py:38]: I'm wondering the same

**Pull request merged**